### PR TITLE
Pass a context including Random object to IAction.Execute() method

### PR DESCRIPTION
**Pull request by hanc1208**
It implements #49.

Context includes Random object.
Seed of Random object is generated using block hash and extra seed.
Extra seed is incremented for each transaction starting at 0.
Thus, each action uses same random seed, but each transaction will use a different random seed.

### CODE REVIEW

File: Libplanet/Action/IContext.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+using System;
+
+namespace Libplanet.Action
+{
+    public class IContext
+    {
+        public Random Random { get; protected set; }
```
Comment (longfin) [inline Libplanet/Action/IContext.cs]: Do we need to provide a setter?

File: Libplanet/Action/Context.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+using System;
+using System.Runtime.CompilerServices;
+using Libplanet.Blocks;
+
+[assembly: InternalsVisibleTo("Libplanet.Tests")]
```
Comment (longfin) [inline Libplanet/Action/Context.cs]: Minor, but InternalsVisibleTo is a global setting so you do not have to declare it again.

File: Libplanet/Action/IContext.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+using System;
+
+namespace Libplanet.Action
+{
+    public class IContext
```
Comment (longfin) [inline Libplanet/Action/IContext.cs]: `IContext` seems a little confusing name. Is `IExecutionContext` too messy?

Comment (dahlia) [reply Libplanet/Action/IContext.cs]: I suggest `IActionContext` for an alternative name.

Comment (dahlia) [reply Libplanet/Action/IContext.cs]: It should be `interface` instead of `class`.

**Pull request merged**